src/components/entity/ha-state-label-badge.js

```diff
@@ -84,7 +84,9 @@ export default new Polymer({
       case 'scene':
       case 'updater':
       case 'script':
-        return domainIcon(state.domain, state.state);
+	// XXX - This is a hack. Is it safe to convert the calls to this in this file
+	// to just pass state and break them out in the function?
+        return domainIcon(state.domain, state.state, state);
```

Comment:
I think that we should split this out in 2 methods: `stateIcon` and `domainIcon`.